engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -164,8 +164,18 @@ object WorkflowExecutionActor {
     }
 
     private def arePrerequisitesDone(key: JobKey): Boolean = {
-      val upstream = key.scope.prerequisiteScopes.toList.map(s => upstreamEntries(key, s))
-      val downstream = key match {
+      val upstream = key.scope match {
+        case node: GraphNode => node.upstream collect {
+          // Only scatters and calls are in the execution store for now (not declarations)
+          // However declarations are nodes so they can be an upstream dependency
+          // We don't want to look for those in the execution store (yet ?) since upstreamEntry would return None
```

Comment:
Is that right? I thought you could `upstream` on a declaration to get all the things that it depends on?